lib/file-source/src/file_server.rs

```diff
@@ -160,10 +158,6 @@ where
                 .instrument(trace_span!("writing checkpoints file"))
```

Comment:
I meant to ask about the actual checkpoint write here. Why is it being spawned in a task, when the caller just `await`s on it anyways?

Reply:
I believe that's just to mark the thread as blocking so that we don't starve the normal runtime threads. It's likely not a big deal in this specific circumstances, but technically correct since we're spawning this as a task.